File: C++/Hard/WrongAnswer/handling-sum-queries-after-update_3.cpp
class Solution {
public:
    vector<long long> handleQuery(vector<int>& nums1, vector<int>& nums2, vector<vector<int>>& queries) {
        
        vector<long long> ans;
        int n = nums1.size();
        vector<long long> sum(n, 0);
        for (int i = 0; i < n; i++) {
            sum[i] = nums2[i];
            if (i) sum[i] += sum[i - 1];
        }
        for (auto& q : queries) {
            if (q[0] == 1) {
                for (int i = q[1]; i <= q[2]; i++) {
                    nums1[i] ^= 1;
                    if (nums1[i]) {
                        sum[i] += nums2[i];
                    } else {
                        sum[i] -= nums2[i];
                    }
                }
            } else if (q[0] == 2) {
                for (int i = 0; i < n; i++) {
                    nums2[i] += nums1[i] * q[1];
                    if (nums1[i]) {
                        sum[i] += nums2[i];
                    } else {
                        sum[i] -= nums2[i];
                    }
                }
            } else {
                int l = q[1], r = q[2];
                if (l == 0) {
                    ans.push_back(sum[r]);
                } else {
                    ans.push_back(sum[r] - sum[l - 1]);
                }
            }
        }
        return ans;
    }
};

File: Java/Hard/WrongAnswer/handling-sum-queries-after-update_3.java
class Solution {
    public long[] handleQuery(int[] nums1, int[] nums2, int[][] queries) {
        
        int n = nums1.length;
        long[] res = new long[queries.length];
        long[] sum = new long[n + 1];
        for (int i = 0; i < n; i++) {
            sum[i + 1] = sum[i] + nums2[i];
        }
        int[] count = new int[n + 1];
        int idx = 0;
        for (int[] query : queries) {
            int type = query[0];
            if (type == 1) {
                int l = query[1], r = query[2];
                count[l]++;
                count[r + 1]--;
            } else if (type == 2) {
                int p = query[1];
                for (int i = 1; i <= n; i++) {
                    count[i] += count[i - 1];
                    sum[i] += count[i] * p;
                }
                Arrays.fill(count, 0);
            } else {
                int l = query[1], r = query[2];
                res[idx++] = sum[r + 1] - sum[l];
            }
        }
        return res;
    }
}